### PR TITLE
#279 json assertions

### PR DESCRIPTION
**Pull request by kqmpetenz**
if json compare fails an attachment displaying the difference between the json files is added to the allure report

### CODE REVIEW

File: src/main/java/com/xceptance/neodymium/util/JsonAssert.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                      + "detectMove: true"
+                    + "}"
+                  + "});"
+
+                  + "const left = " + expectedJson + ";"
+                  + "const right = " + actualJson + ";"
+                  
+                  + "function sortArraysInJson(json) {"
+                    + "for (const key in json) {"
+                      + "if (Array.isArray(json[key])) {"
```
Comment (wurzelkuchen) [inline src/main/java/com/xceptance/neodymium/util/JsonAssert.java:57]: this crashes when an empty array is part of the json. please change to `if (Array.isArray(json[key]) && json[key].lenght>0) {`

File: src/main/java/com/xceptance/neodymium/util/JsonAssert.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                + "</script>"
+              + "</body>"
+            + "</html>";
+        
+        return html;
+    }
+
+    public static void assertEquals(String expectedJson, String actualJson, JSONCompareMode jsonCompareMode)
+    {
+        System.out.println(jsonCompareMode.name());
```
Comment (wurzelkuchen) [inline src/main/java/com/xceptance/neodymium/util/JsonAssert.java:115]: that should probably not be there

**Review by wurzelkuchen — CHANGES_REQUESTED** [review]
need a quick fix, otherwise fine

File: src/main/java/com/xceptance/neodymium/util/JsonAssert.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
+package com.xceptance.neodymium.util;
+
+import org.json.JSONException;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import io.qameta.allure.Allure;
+
+public class JsonAssert
```
Comment (wurzelkuchen) [inline src/main/java/com/xceptance/neodymium/util/JsonAssert.java:9]: This class needs javadoc

**Review by wurzelkuchen — CHANGES_REQUESTED** [review]
forgot about that one

File: src/main/java/com/xceptance/neodymium/util/JsonAssert.java (hunk 1 above)
Comment (kqmpetenz) [reply src/main/java/com/xceptance/neodymium/util/JsonAssert.java:57]: done (ist aber .length ;) )

File: src/main/java/com/xceptance/neodymium/util/JsonAssert.java (hunk 2 above)
Comment (kqmpetenz) [reply src/main/java/com/xceptance/neodymium/util/JsonAssert.java:115]: done

File: src/main/java/com/xceptance/neodymium/util/JsonAssert.java (hunk 3 above)
Comment (kqmpetenz) [reply src/main/java/com/xceptance/neodymium/util/JsonAssert.java:12]: done

**Pull request merged**